java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture7/tasks/_2/ThreadOperations.java

```diff
@@ -0,0 +1,112 @@
+package com.zoolatech.lecture7.tasks._2;
+
+class ThreadOperations {
+
+    public static void explicitLocks() throws InterruptedException {
+        Calculator calculator = new Calculator(5);
+
+        Runnable taskAddition = () -> {
+            for (int i = 0; i < 1000; i++) {
+                calculator.additioinExplicitLock(3);
+            }
+            System.out.println("Current Thread: " + Thread.currentThread().getName());
+        };
+
+        Runnable taskSubstraction = () -> {
+            for (int i = 0; i < 1000; i++) {
+                calculator.subtractionExplicitLock(3);
+            }
+            System.out.println("Current Thread: " + Thread.currentThread().getName());
+        };
+
+        Thread[] threads = new Thread[10];
+        for (int i = 0; i < threads.length; i++) {
+            threads[i] = new Thread(taskAddition);
+            threads[i].start();
+        }
+        for (Thread thread : threads) {
+            thread.join();
+        }
+        Thread.sleep(1000);
+        System.out.println("\n" + calculator.currentValue() + " Current thread: " + Thread.currentThread().getName() + "\n");
+        Thread.sleep(1000);
+
+        for (int i = 0; i < threads.length; i++) {
```

Comment:
threads creation can be moved to a private method (e.g., `runTask`), to which you pass an instance of the task:
```
runTask(taskAddition);
runTask(taskSubstraction);